docs/using-the-compiler.rst

```diff
@@ -287,8 +287,15 @@ Input Description
               // Improve allocation of stack slots for variables, can free up stack slots early.
               // Activated by default if the Yul optimizer is activated.
               "stackAllocation": true,
-              // Select optimization steps to be applied.
-              // Optional, the optimizer will use the default sequence if omitted.
+              // Select optimization steps to be applied. It is also possible to modify both the
+              // optimization sequence and the clean-up sequence. Instructions for each sequence
+              // are separated with the : delimiter. For more information see
+              // "The Optimizer > Selecting Optimizations".
+              // This field is optional and if not provided, the default sequences for both
+              // optimization and clean-up are used. If only one of the options is provivded
+              // the other will not run unless it is also provided.
+              // If set to an empty value, only the default clean-up sequence is used and
+              // no optimization steps are applied.
```

Comment:
[Here](https://github.com/ethereum/solidity/pull/13650#discussion_r1010246383), @nikola-matic said that, when set to an empty non delimited sequence, both optimizer and clean-up default are executed. Which one is it?

Reply:
So I had to remake this PR however in the original Cameel stated: "omitting the whole field will make the compiler use defaults for both sequences but setting it to an empty value will not - it will instead make the compiler use the default cleanup sequence. I.e. to clear both you set it to `:`" Which is what I was trying to convey in the text.

Original PR: https://github.com/ethereum/solidity/pull/13619

Reply:
Ok. So we need to add that too. Considering the order in which we present the information, I suggest the following:
1. Field is optional
2. Field not provided, then default sequences for both optimization and clean-up
3. If one option value is empty, it will not execute only the other one.
4. --> If both values are empty, i.e, ":", then neither optimization nor clean-up steps are executed. <-- Should be added here, IMO.
5. If value is an empty string, i.e, "", default clean-up sequence only is used.

Please, keep the text that is already present and add a sentence of your own detailing the case with delimiter `:` making both values empty.

Reply:
That being said, I am thinking that we should also specify to what the first value is referring to, optimization or clean-up?

Reply:
Added some updates here based on your suggestions. Let me know what you think.